generator/lib/code_builders/assignment_builder.dart

```diff
@@ -46,46 +54,204 @@ Expression generateSourceFieldAssignment(SourceAssignment sourceAssignment,
-            .call([sourceFieldAssignment]);
+        sourceFieldAssignment = invokeNestedMappingFunction(
+          matchingMappingMethods.first, 
+          sourceAssignment.refChain!.isNullable,
+          refer(sourceAssignment.refChain!.refWithQuestion),
+          refer(sourceAssignment.refChain!.ref),
+        );
       }
     }
   }
   return sourceFieldAssignment;
 }
 
+Expression invokeNestedMappingFunction(
+  MethodElement method, 
+  bool sourceNullable,
+  Expression refWithQuestion,
+  Expression ref,
+) {
+  Expression sourceFieldAssignment;
+  if(method.parameters.first.isOptional) {
+    // The parameter can be null.
+    sourceFieldAssignment = refer(method.name)
+        .call([refWithQuestion]);
+  } else {
+    sourceFieldAssignment = refer(method.name)
+        .call([ref]);
+    sourceFieldAssignment = checkNullExpression(
+      sourceNullable,
+      refWithQuestion, 
+      sourceFieldAssignment
+    );
+  }
+  return sourceFieldAssignment;
+}
+
+Expression invokeNestedMappingForStaticFunction(
+  ExecutableElement sourceFunction,
+  ClassElement abstractMapper,
+  VariableElement targetField,
+  Expression sourceFieldAssignment,
+) {
+  final returnType = sourceFunction.returnType;
+  final matchingMappingMethods = _findMatchingMappingMethod(
+      abstractMapper, targetField.type, returnType);
+  if(matchingMappingMethods.isNotEmpty) {
+    final nestedMappingMethod = matchingMappingMethods.first;
+
+    if(
+      nestedMappingMethod.parameters.first.type.nullabilitySuffix != NullabilitySuffix.question &&
+      sourceFunction.returnType.nullabilitySuffix == NullabilitySuffix.question
+    ) {
+      final str = makeNullCheckCall(
+        sourceFieldAssignment.accept(
+          DartEmitter()
+        ).toString(),
+        nestedMappingMethod,
+      );
+      sourceFieldAssignment = refer(str);
+    } else {
+      sourceFieldAssignment = refer(matchingMappingMethods.first.name)
+          .call([sourceFieldAssignment]);
+    }
+
+  }
+  return sourceFieldAssignment;
+}
+
 /// Finds a matching Mapping Method in [classElement]
 /// which has the same return type as the given [targetReturnType] and same parametertype as the given [sourceParameterType]
 Iterable<MethodElement> _findMatchingMappingMethod(ClassElement classElement,
     DartType targetReturnType, DartType sourceParameterType) {
   final matchingMappingMethods = classElement.methods.where((met) {
-    return met.returnType == targetReturnType &&
-        met.parameters.isNotEmpty && met.parameters.first.type == sourceParameterType;
+    // Sometimes the user is troubled by the nullability of these types.
+    // So ingore the nullability of all the type for the nested mapping function is more easy to be matched.
+    // The process of nullability is one duty for this library.
+    final metReturnElement = met.returnType.element;
+    final metParameterElement = met.parameters.first.type.element;
```

Comment:
There seems to be a problem with the StaticMappingMapper Test here

Reply:
Thank you for your reply. I must apologize for that I didn't run the unit tests of this project. I will do some work to pass these tests. I didn't know how to test the code that use source_gen, until I saw these unit tests code in this project. I'm glad to have acquired this new knowledge.

For this point:
> Pass on the mapper pointer to the function mapping method by the named parameters.

The idea is from here: 

```dart

class Source{
    final subList = ToMany<SubSource>();
    final one = ToOne<SubSource>();
}

class SubSource {}

class Target {
    late List<SubTarget> subList;
    late SubTarget one; 
}

class SubTarget {}

@Mapper()
abstract class ObjectMapper {
    @Mapping(source: mapToManySubSource, target: "list")
    @Mapping(source: mapToOneSubSource, target: "list")
    Target fromSource(Source source);

    SubTarget fromSubSource(SubSource source);

}

List<SubTarget> mapToManySubSource(ToMany<SubSource> toMany, {required ObjectMapper mapper}) {
    return toMany.map((x) => mapper.fromSubSource(x)).toList();
}

SubTarget mapToOneSubSource(ToOne<SubSource> one, {required ObjectMapper mapper}) {
    return mapper.fromSubSource(one.target!);
}
```

This is first intention I want to pass on the mapper to the function mapping method.
1. The ToMany is a class implemented the List. And then I extends the availability of the shouldAssignList, the ToMany can be mapped automatically. 

2. Then I try to apply the nested mapping after the function mapping. When they work together. I can map the "ToOne\<SubSource\>" too.

The function mapping "mapToManySubSource" and "mapToOneSubSource" are not needed now. So the reason I need to pass on the mapper no longer exists. 

But I think there is still some complex container and custom container, which can't be mapped automatically. We still have to handle them with function mapping method and need the mapper object to map the source item in the source container to the target item which in the target container.For example, on the source side, the container object which not implement "List" or "Iterable"  is not supported. And  on the target side only the basic container: "List", "Iterable", "Set" is supported. 

```dart
class CustomList<T> {
    
    final List<T> value;

    CustomList(this.value);
}

class Source{
    late final List<SubSource> list;
    // Map can't be mapped from automatically.
    late final Map<String, SubSource> map;
}

class SubSource {}

class Target {
    // CustomList can't be mapped to automatically.
    late final CustomList<SubTarget> list;
    late final Map<String, SubTarget> map;
}

class SubTarget {}


@Mapper()
abstract class ObjectMapper {
    Target fromSource(Source source);
    SubTarget fromSubSource(SubSource source);
}

``` 
We need handle the Map and CustomList with function mapping. But we can't do it without mapper object.

```dart

mapMap(Map<String, SubSource> map, {required ObjectMapper mapper}) {
    return map.map((k, v) => MapEntry<String, SubTarget>(k, mapper.fromSubSource(v)));
}
mapList(List<SubSource> list, {required ObjectMapper mapper}>) {
    return CustomList<SubTarget>(list.map((x) => mapper.fromSubSource(x)).toList());
}
```

The usecase is just my imagination. My work doesn't in this case. I just thought the function mapping with the mapper object is more powerful. It can deal with the situations that cannot be predicted.